### DISCUSSION TITLE
issues about self hosted

### DISCUSSION THREAD

**Discussion by dfang** [supabase/supabase#13570] [Questions] [answered]
Hi,  Supabase is a very cool project, i pay close attention to it from it's born.  
but I'm in China, for my job i can only use self hosted version if using supabase.

1.  for storage, does supabase support custom storage provider, aliyun/tencent object storage service, or self hosted minio?  i searched issues, docs and discussions, it seems support for custom storage providers is lacking, or docs missing.

2. for auth, does supabase support custom auth provider (eg. wechat, lark), is there a doc on how to implement that.

3. for logs i can use vector to send docker logs to loki, but for analytics, logflare using google big query, is there a plan b?

4. for edge functions, i'm not sure deno deploy has China endpoints, and if we use supabase in private network, is there any alternatives?

thanks !

**Comment by joepa37** [comment]
For Edge Functions they released [Edge Runtime](https://github.com/supabase/edge-runtime) for local deployments, so I assume they will give deep understanding about this on the [Launch Week 7](https://supabase.com/launch-week)

**Comment by Ziinc** [comment, marked as answer]
1. AWS is officially supported, if support for s3 compatible options is mission-critical, you may consider contributing to the [storage-api](https://github.com/supabase/storage-api) repo for this feature. It is currently being considered but is not a priority for the team at the moment.
2. likewise, for additional oauth2 providers not under this [list](https://github.com/supabase/gotrue#external-authentication-providers), the team would definitely consider other widely used providers not in the list.
3. if self-hosting, unfortunately Logs and Reports require BigQuery. We are working to remove this dependency.
4. For a self-hosted runtime, you can consider the [supabase/edge-runtime project](https://github.com/supabase/edge-runtime) if you are unable to use providers outside of your vpn.
